### DISCUSSION TITLE
Possible to download a single patch representing all closed issues of a milestone?

### DISCUSSION THREAD

**Discussion by graysky2** [community/community#22306] [New to GitHub]
<p>One can append a “.patch” to a pull request’s URL to have github generate a patch which is really convenient.  Is it possible to have github generate a single patch for all closed issues/merged PRs associated with a project milestone somehow?  For example, there are currently 19 closed PRs associated with a <a href="https://github.com/xbmc/xbmc/milestone/121?closed=1" rel="nofollow noopener">Leia milestone</a> … can I get github to make a single patch for these 19?  Alternatively, I can manually go through them and make github generate the corresponding patches as I described above.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Interesting question.</p>
<p>There isn’t a way that I know of to do this through the GitHub UI. Because GitHub is relying on Git functionality to generate the patch that you see, you might be able to build a tool that could query the GitHub API for the information about pull requests associated with a milestone and then generate the patch yourself from a local clone of the repository.</p>
<p>I hope that helps! Let us know if you have more questions.</p>

**Comment by graysky2** [comment]
<p>Thank you for the reply.  I have a hacky bash script that lists out each indiviual patch for inclusion in a package, but doing it via an API would be must slicker.</p>
<pre><code>#!/bin/bash
# array of closed issues associated with a milestone

PR=(
  15483
  15521
  15535
  15434
  15479
  15489
  15480
  15556
  15516
  15553
  15499
  15195
  15543
  15557
  15574
  15447
  15586
  15580
  15595
  15600
  15602
  14665
  15619
)

for ((i = 0; i &lt; ${#PR[@]}; ++i)); do
  position=$(( $i + 1 ))
  padded=$(printf "%02d" $position)
  echo " $padded-PR${PR[$i]}.patch::https://patch-diff.githubusercontent.com/raw/xbmc/xbmc/pull/${PR[$i]}.patch"
done
</code></pre>
